lib/pretty-repl.js

```diff
@@ -191,58 +229,21 @@ class PrettyREPLServer extends repl.REPLServer {
     this.output.write(this._prompt + stringToWrite);
   }
 
-  _insertString (c) {
+  _insertString(c) {
     this.lineBeforeInsert = this.line;
     try {
-      return super._insertString (c);
+      return super._insertString(c);
     } finally {
       this.lineBeforeInsert = undefined;
     }
   }
 
-  _doColorize = memoizeStringTransformerMethod(100, function(str) {
-    return this.colorize(str);
-  });
-
-  _stripCompleteJSStructures = memoizeStringTransformerMethod(1000, function(str) {
-    // Remove substructures of the JS input string `str` in order to simplify it,
-    // by removing matching pairs of quotes and parentheses/brackets.
-
-    // Specifically, remove all but the last, non-nested pair of (), because ()
-    // can affect whether the previous word is seen as a keyword.
-    // E.g.: When input is `function() {`, do not replace the ().
-    //       When input is `{ foo(); }`, do replace the `()`, then afterwards the `{ ... }`.
-    const brackets = this._findAllMatchingBracketsIgnoreMismatches(str);
-    if (brackets.length > 0) {
-      const last = brackets[brackets.length - 1];
-      if (last.kind === '(' && (last.parent == null || last.parent.end === -1))
-        brackets.pop();
-    }
-    // Remove brackets in reverse order, so that their indices remain valid.
-    for (let i = brackets.length - 1; i >= 0; i--) {
-      // Skip brackets which are contained in outer brackets that will also
-      // be removed.
-      if (brackets[i].parent !== null && brackets[i].parent.end !== -1) continue;
-      str = str.substr(0, brackets[i].start) + str.substr(brackets[i].end + 1);
-    }
-    return str;
-  });
-
-  _findAllMatchingBracketsIgnoreMismatches = memoizeStringTransformerMethod(1000, function(str) {
-    return findAllMatchingBrackets(str, true);
-  });
-  _findAllMatchingBracketsIncludeMismatches = memoizeStringTransformerMethod(1000, function(str) {
-    return findAllMatchingBrackets(str, false);
-  });
```

Comment:
Similarly here … did the linter have an issue with the class field assignments? That’s a normal JS feature that we should be able to use, I think

Reply:
yep no idea why it doesn't like class properties even if i set ES2022 with parserOptions, at some point i gave up